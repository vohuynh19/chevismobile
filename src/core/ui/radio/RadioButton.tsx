import React, {ComponentProps} from 'react';

import {makeStyles} from '../../theme';

import {CheckBox} from '../check-box/CheckBox';

export const RadioButton = (props: ComponentProps<typeof CheckBox>) => {
  const styles = useStyles();

  return (
    <CheckBox testID="btnRadio" {...props} checkboxStyle={styles.checkBox} />
  );
};

const useStyles = makeStyles({
  checkBox: {
    borderRadius: 'full',
  },
});
