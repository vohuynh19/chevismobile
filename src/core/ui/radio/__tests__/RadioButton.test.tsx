import React from 'react';
// eslint-disable-next-line no-restricted-imports
import {View} from 'react-native';

import {fireEvent, render, screen} from 'testUtils';

import {RadioButton} from '../RadioButton';

describe('RadioButton', () => {
  test('should render component', async () => {
    await render(<RadioButton checked onChange={() => {}} />, {
      withSnapshot: true,
    });

    // Disable until we upgrade react-native-reanimated
    // const checkmark = container.findAllByType(Animated.View)[0]; // there are two Animated.View, first is checkbox, second is checkmark

    // expect(checkmark).toHaveStyle({ borderRadius: 9999 });
  });

  // Disable until we upgrade react-native-reanimated
  test.skip('should show checkmark when checked is true', async () => {
    const {container} = await render(
      <RadioButton checked onChange={() => {}} />,
    );

    const viewIconComponent = container.findAllByType(View)[2];
    // Animated.View is react-native's View with animation. Pressable also is react-native's View in unit test.
    // There are three View, first is pressable, second is border of checkbox. third is checkmark

    expect(viewIconComponent).toHaveStyle({opacity: 1});
  });

  // Disable until we upgrade react-native-reanimated
  test.skip('should not show checkmark when checked is false', async () => {
    const {container} = await render(
      <RadioButton checked={false} onChange={() => {}} />,
    );

    const viewIconComponent = container.findAllByType(View)[2];
    // Animated.View is react-native's View with animation. Pressable also is react-native's View in unit test.
    // There are three View, first is pressable, second is border of checkbox. third is checkmark

    expect(viewIconComponent).toHaveStyle({opacity: 0});
  });

  test('should invoke onChange function with param is true if user presses radiobutton and checked variable is true', () => {
    const mockedHandleChange = jest.fn();
    render(<RadioButton checked onChange={mockedHandleChange} />);
    fireEvent.press(screen.getByRole('checkbox'));

    expect(mockedHandleChange).toBeCalledWith(false);
    expect(mockedHandleChange).toBeCalledTimes(1);
  });

  test('should invoke onChange function with param is false if user presses radiobutton and checked variable is false', () => {
    const mockedHandleChange = jest.fn();
    render(<RadioButton checked={false} onChange={mockedHandleChange} />);
    fireEvent.press(screen.getByRole('checkbox'));

    expect(mockedHandleChange).toBeCalledWith(true);
    expect(mockedHandleChange).toBeCalledTimes(1);
  });
});
