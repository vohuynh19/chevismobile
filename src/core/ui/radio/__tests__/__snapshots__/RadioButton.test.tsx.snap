// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton should render component 1`] = `
<View
  accessibilityRole="checkbox"
  accessibilityState={
    {
      "busy": undefined,
      "checked": true,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
  testID="btnRadio"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 4,
          "borderWidth": 2,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        {
          "backgroundColor": undefined,
          "borderColor": undefined,
        },
        {
          "borderRadius": 9999,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        {
          "opacity": undefined,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        fill="#FBFAFF"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={16}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294703871,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M18.207 7.293a1 1 0 0 1 0 1.414l-8 8a1 1 0 0 1-1.414 0l-3-3a1 1 0 1 1 1.414-1.414L9.5 14.586l7.293-7.293a1 1 0 0 1 1.414 0Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
